pkg/3scale/amp/component/backend_options.go

```diff
@@ -0,0 +1,98 @@
+package component
+
+import (
+	oprand "github.com/3scale/3scale-operator/pkg/crypto/rand"
+	"github.com/go-playground/validator/v10"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+type BackendOptions struct {
+	ServiceEndpoint              string `validate:"required"`
+	RouteEndpoint                string `validate:"required"`
+	StorageURL                   string `validate:"required"`
+	QueuesURL                    string `validate:"required"`
```

Comment:
Regarding the "required" struct tag field of the Validator:

When having a field as "required" in the new validator context, if I understand correctly, specifying the "required" tag it makes sure that the set value for that option is NOT the zero-value.
How would we handle a required value, that we want to be sure is specified in the struct but at the same time we want to be able to allow the zero-value for it?  Kubernetes tells you to use pointers when using optional values. To distinguish the zero-value from a value not being introduced. Would we use a pointer too in that case?

Reply:
see comment above